core/domain/activity_jobs_one_off.py

```diff
@@ -471,3 +478,129 @@ def map(snapshot_model):
+        # or delete.
+        rights_model = [
+            'CollectionRightsSnapshotMetadataModel',
+            'ExplorationRightsSnapshotMetadataModel'
+        ]
+
+        # Note: The subtopic snapshot ID is in the format
+        # '<topicId>-<subtopicNum>-<version>'.
+        model_id, version = snapshot_model.id.rsplit('-', 1)
+        model_dict = {
+            'CollectionSnapshotMetadataModel': {
+                'parent_model': collection_models.CollectionModel,
+                'commit_log_model': (
+                    collection_models.CollectionCommitLogEntryModel),
+                'id_string_format': 'collection-%s-%s'
+            },
+            'ExplorationSnapshotMetadataModel': {
+                'parent_model': exp_models.ExplorationModel,
+                'commit_log_model': exp_models.ExplorationCommitLogEntryModel,
+                'id_string_format': 'exploration-%s-%s'
+            },
+            'QuestionSnapshotMetadataModel': {
+                'parent_model': question_models.QuestionModel,
+                'commit_log_model': question_models.QuestionCommitLogEntryModel,
+                'id_string_format': 'question-%s-%s'
+            },
+            'SkillSnapshotMetadataModel': {
+                'parent_model': skill_models.SkillModel,
+                'commit_log_model': skill_models.SkillCommitLogEntryModel,
+                'id_string_format': 'skill-%s-%s'
+            },
+            'StorySnapshotMetadataModel': {
+                'parent_model': story_models.StoryModel,
+                'commit_log_model': story_models.StoryCommitLogEntryModel,
+                'id_string_format': 'story-%s-%s'
+            },
+            'TopicSnapshotMetadataModel': {
+                'parent_model': topic_models.TopicModel,
+                'commit_log_model': topic_models.TopicCommitLogEntryModel,
+                'id_string_format': 'topic-%s-%s'
+            },
+            'SubtopicPageSnapshotMetadataModel': {
+                'parent_model': subtopic_models.SubtopicPageModel,
+                'commit_log_model': (
+                    subtopic_models.SubtopicPageCommitLogEntryModel),
+                'id_string_format': 'subtopicpage-%s-%s'
+            },
+            'TopicRightsSnapshotMetadataModel': {
+                'parent_model': topic_models.TopicRightsModel,
+                'commit_log_model': topic_models.TopicCommitLogEntryModel,
+                'id_string_format': 'rights-%s-%s'
+            },
+            'CollectionRightsSnapshotMetadataModel': {
+                'parent_model': collection_models.CollectionRightsModel,
+                'commit_log_model': (
+                    collection_models.CollectionCommitLogEntryModel),
+                'id_string_format': 'rights-%s-%s'
+            },
+            'ExplorationRightsSnapshotMetadataModel': {
+                'parent_model': exp_models.ExplorationRightsModel,
```

Comment:
parent_model_class, commit_log_model_class

(distinguish clearly between the "model class" and a "model" which is an instance of the class)

Reply:
Done.